File: 231221/거품 정렬 구현/implement-bubble-sort.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int n = Integer.parseInt(br.readLine());
        List<Integer> arr = new ArrayList<>();
        
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            // Integer.parseInt를 사용하여 문자열을 정수로 변환해야 합니다.
            arr.add(Integer.parseInt(st.nextToken()));
        }
        List<Integer> sortedArr = bubblSort(arr);
        for(int i : sortedArr){
            System.out.print(i + " ");
        }
    }
  private static List<Integer> bubblSort(List<Integer> arr) {
        int n = arr.size();
        for (int i = 0; i < n - 1; i++) {
            boolean isSwap = false;
            for (int j = 0; j < n - i - 1; j++) {
                if (arr.get(j) > arr.get(j + 1)) {
                    int temp = arr.get(j);
                    arr.set(j, arr.get(j + 1));
                    arr.set(j + 1, temp);
                    isSwap = true;
                }
            }

            if (!isSwap) {
                break;
            }
        }
        return arr;
    }
}

File: 231221/거품 정렬 구현/implement-bubble-sort.py
n = int(input())
arr = list(map(int, input().split()))

def bubbl_sort(arr: list[int]) -> list[int]:
    for i in range(n - 1):
        swapped = False
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True

        if not swapped:
            break
    return arr

print(*bubbl_sort(arr))